File: cpp/session/UserPassMismatchException.cpp
#include "session/UserPassMismatchException.hpp"
#include "common/ErrorPkg.hpp"

const std::string UserPassMismatchException::makeMessage(
		const std::string &clID, const std::string &pw)
{
	std::vector<std::string> args;
	args.push_back("<<clID>>");
	args.push_back("<<pw>>");

	std::vector<std::string> vals;
	vals.push_back(clID);
	vals.push_back(pw);

	return ErrorPkg::getMessage("epp.login.fail.auth.pw", args, vals);
}

File: src/main/java/com/ausregistry/jtoolkit2/session/UserPassMismatchException.java
package com.ausregistry.jtoolkit2.session;

import com.ausregistry.jtoolkit2.ErrorPkg;

/**
 * The credentials supplied to the EPP login command were considered invalid
 * by the EPP server to which the login was sent.  Intervention is required to
 * either change the password or supply the correct credentials to the login
 * command.
 */
public class UserPassMismatchException extends LoginException {
    /**
     *
     */
    private static final long serialVersionUID = -5500476048792038560L;
    private static final String[] USER_PASS_ARR = new String[] {
        "<<clID>>", "<<pw>>"
    };

    public UserPassMismatchException() {
        super("Incorrect password for specified user");
    }

    public UserPassMismatchException(String msg) {
        super(msg);
    }

    public UserPassMismatchException(Throwable cause) {
        super(cause);
    }

    public UserPassMismatchException(String clID, String pw) {
        super(ErrorPkg.getMessage("epp.login.fail.auth.pw",
                    USER_PASS_ARR,
                    new String[] {clID, pw}));
    }
}

